azure-pipelines-product-construction-service.yml

```diff
@@ -0,0 +1,67 @@
+# Changes the format of the Build.BuildNumber variable
+name: $(Date:yyyyMMdd)$(Rev:r)
+
+trigger:
+  branches:
+    exclude:
+      - production
+
+variables:
+- name: resourceGroupName
+  value: 'product-construction-service'
+- name: containerName
+  value: 'product-construction-service.api'
+- ${{ if ne(variables['Build.SourceBranch'], 'refs/heads/production') }}:
+  - group: 'Product-Construction-Service-Int'
+  - name: containerappName
+    value: product-construction-int
+  - name: containerRegistryName
+    value: productconstructionint
+  - name: subscriptionName
+    value: HelixStaging
+  - name: containerappEnvironmentName
+    value: product-construction-service-env-int
+  - name: dockerRegistryUrl
+    value: productconstructionint.azurecr.io
+
+stages:
+- stage: PublishProductConstructionService
+  displayName: Publish Product Construction Service
+  dependsOn: []
+  jobs:
+  - job: Publish
+    displayName: Publish Product Construction Service
+    pool:
+      vmImage: ubuntu-latest
+      # name: NetCore1ESPool-Internal
+      # demands: ImageOverride -equals 1es-ubuntu-2004
+
+
+    steps:
+    - checkout: self
+
+    - powershell: |
+        $shortSha = "$(Build.SourceVersion)".Substring(0,10)
+        $newDockerTag = "$shortSha-$(Build.BuildNumber)"
+        Write-Host "##vso[task.setvariable variable=newDockerImageTag]$newDockerTag"
+        Write-Host "set newDockerImageTag to $newDockerTag"
+      displayName: 'Set new docker image tag variable'
+
+    - powershell: docker build . -f $(Build.SourcesDirectory)/src/ProductConstructionService/ProductConstructionService.Api/Dockerfile -t "$(dockerRegistryUrl)/$(containerName):$(newDockerImageTag)"
+      displayName: 'Build docker image'
+
+    - ${{ if notin(variables['Build.Reason'], 'PullRequest') }}:
```

Comment:
I'd rather this said `if branch is main`

Reply:
if we do that, we won't be able to push to the staging environment from dev branches. I think this is inline with what we have for the main pipeline